src/panels/lovelace/editor/view-editor/hui-view-visibility-editor.ts

```diff
@@ -0,0 +1,156 @@
+import {
+  html,
+  LitElement,
+  TemplateResult,
+  customElement,
+  property,
+  PropertyValues,
+  CSSResult,
+  css,
+} from "lit-element";
+import "@polymer/paper-input/paper-input";
+
+import { HomeAssistant } from "../../../../types";
+import { fireEvent } from "../../../../common/dom/fire_event";
+import { configElementStyle } from "../config-elements/config-elements-style";
+import { LovelaceViewConfig, ShowViewConfig } from "../../../../data/lovelace";
+
+import { fetchUsers, User } from "../../../../data/user";
+import memoizeOne from "memoize-one";
+import { compare } from "../../../../common/string/compare";
+import { HaSwitch } from "../../../../components/ha-switch";
+
+declare global {
+  interface HASSDomEvents {
+    "view-visibility-changed": {
+      visible: ShowViewConfig[];
+    };
+  }
+}
+
+@customElement("hui-view-visibility-editor")
+export class HuiViewVisibilityEditor extends LitElement {
+  set config(config: LovelaceViewConfig) {
+    this._config = config;
+    this._visible = this._config.visible;
+  }
+
+  @property() public hass!: HomeAssistant;
+  @property() public _config!: LovelaceViewConfig;
+  @property() private _users!: User[];
+  @property() private _visible?: boolean | ShowViewConfig[];
+
+  private _sortedUsers = memoizeOne((users: User[]) => {
+    return users
+      .filter((user) => !user.system_generated)
+      .sort((a, b) => compare(a.name, b.name));
+  });
+
+  protected firstUpdated(changedProps: PropertyValues) {
+    super.firstUpdated(changedProps);
+
+    fetchUsers(this.hass).then((users) => {
+      this._users = users;
+      // how can I resize and reposition dialog? Should I use `iron-resize` event?
+      // fireEvent(this as HTMLElement, "iron-resize");
+    });
+  }
+
+  protected render(): TemplateResult {
+    if (!this.hass || !this._users) {
+      return html``;
+    }
+
+    return html`
+      ${configElementStyle}
+      <div class="card-config">
+        <span>Select which users should have access to this view</span>
+
+        ${this._sortedUsers(this._users).map(
+          (user) => html`
+            <div class="flex">
+              <div>${user.name}</div>
+              <ha-switch
+                .userId="${user.id}"
+                @change=${this.valChange}
+                .checked=${this.checkUser(user.id)}
+                >&nbsp;</ha-switch
```

Comment:
I had to add `&nbsp;` to avoid showing unwanted scrollbar

Reply:
Let's fix that in `ha-switch` instead

Reply:
Can you do it? I can remove that character, but until this is fixed in `ha-switch` then the scrollbars will appear.

Reply:
I'll take a look, can you create an issue?

Reply:
@bramkragten issue created: https://github.com/home-assistant/frontend/issues/5084